cisa-advisory/operations.py

```diff
@@ -0,0 +1,228 @@
+import json
+import requests
+from bs4 import BeautifulSoup
+import datetime
+from dateutil.relativedelta import relativedelta
+from connectors.core.connector import get_logger, ConnectorError
+from .constant import *
+from integrations.crudhub import make_request
+
+logger = get_logger('cisa-advisory')
+
+class Advisory():
+
+    def __init__(self, url):
+         self.url = url
+
+    def get_ics_data(self):
+        try:
+            ics_advisory_url_by_year_links_list = []
+            output = []
+            response = requests.get(self.url, verify=False)
+            soup = BeautifulSoup(response.text, 'html.parser')
+            for link in soup.find_all('a'):
+                if 'ics-advisory' in link.get('href'):
+                    ics_advisory_url_by_year_links_list.append(link.get('href'))
+            for advisory_link in ics_advisory_url_by_year_links_list:
+                json_file_data = requests.get(self.url + advisory_link, verify=False)
+                for item in json.loads(json_file_data.text):
+                    output.append(item)
+            return output
+        except Exception as err:
+            logger.exception(str(err))
+            raise ConnectorError(err)
+
+    def date_filter_advisory(self, params, ics_data):
+        try:
+            today_date = datetime.date.today()
+            if params['date_filter'] == 'Today':
```

Comment:
Can you make a dictionary with date filter value as key and its corresponding days as its value this will remove multiple if statements and will also be in O(1) 

Eg =  {"Last 7 Days":-7,
           "Last 30 Days":-30}
           
 days = Eg.get(params.get('date_filter'))
filter_date = today_date + relativedelta(days=days)

Reply:
As discussed we can not incorporate it in dict